undefined reference if `TESTING_ENABLE=ON`

**Bug Description**

```
make build CMAKE_BUILD_TYPE=Debug TESTING_ENABLE=ON
```

build log
```
libbase_test.a(sql_sdk_base_test.cc.o): In function `openmldb::sdk::SQLSDKTest::CreateTables(hybridse::sqlcase::SqlCase&, std::shared_ptr<openmldb::sdk::SQLRouter>, int)':
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:65: undefined reference to `hybridse::sqlcase::SqlCase::BuildCreateSqlFromInput(int, std::string*, int) const'
libbase_test.a(sql_sdk_base_test.cc.o): In function `openmldb::sdk::SQLSDKTest::CreateProcedure(hybridse::sqlcase::SqlCase&, std::shared_ptr<openmldb::sdk::SQLRouter>, bool)':
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:121: undefined reference to `hybridse::sqlcase::SqlCase::ExtractTableDef(std::vector<std::string, std::allocator<std::string> > const&, std::vector<std::string, std::allocator<std::string> > const&, hybridse::type::TableDef&)'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:123: undefined reference to `hybridse::sqlcase::SqlCase::BuildCreateSpSqlFromSchema(hybridse::type::TableDef const&, std::string const&, std::set<unsigned long, std::less<unsigned long>, std::allocator<unsigned long> > const&, std::string*)'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:127: undefined reference to `hybridse::sqlcase::SqlCase::BuildCreateSpSqlFromInput(int, std::string const&, std::set<unsigned long, std::less<unsigned long>, std::allocator<unsigned long> > const&, std::string*)'
libbase_test.a(sql_sdk_base_test.cc.o): In function `openmldb::sdk::SQLSDKTest::InsertTables(hybridse::sqlcase::SqlCase&, std::shared_ptr<openmldb::sdk::SQLRouter>, openmldb::sdk::InsertRule)':
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:206: undefined reference to `hybridse::sqlcase::SqlCase::BuildInsertSqlListFromInput(int, std::vector<std::string, std::allocator<std::string> >*) const'
libbase_test.a(sql_sdk_base_test.cc.o): In function `openmldb::sdk::SQLSDKTest::BatchExecuteSQL(hybridse::sqlcase::SqlCase&, std::shared_ptr<openmldb::sdk::SQLRouter>, std::vector<std::string, std::allocator<std::string> > const&)':
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:305: undefined reference to `hybridse::sqlcase::SqlCase::ExtractParameterTypes() const'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:311: undefined reference to `hybridse::sqlcase::SqlCase::ExtractRows(google::protobuf::RepeatedPtrField<hybridse::type::ColumnDef> const&, std::vector<std::vector<std::string, std::allocator<std::string> >, std::allocator<std::vector<std::string, std::allocator<std::string> > > > const&, std::vector<hybridse::codec::Row, std::allocator<hybridse::codec::Row> >&)'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:334: undefined reference to `hybridse::sqlcase::SqlCase::ExtractOutputSchema(hybridse::type::TableDef&) const'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:339: undefined reference to `hybridse::sqlcase::SqlCase::ExtractOutputData(std::vector<hybridse::codec::Row, std::allocator<hybridse::codec::Row> >&) const'
libbase_test.a(sql_sdk_base_test.cc.o): In function `openmldb::sdk::SQLSDKQueryTest::RequestExecuteSQL(hybridse::sqlcase::SqlCase&, std::shared_ptr<openmldb::sdk::SQLRouter>, bool, bool, bool)':
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:407: undefined reference to `hybridse::sqlcase::SqlCase::ExtractInputTableDef(hybridse::type::TableDef&, int) const'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:408: undefined reference to `hybridse::sqlcase::SqlCase::ExtractInputData(std::vector<hybridse::codec::Row, std::allocator<hybridse::codec::Row> >&, int) const'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:409: undefined reference to `hybridse::sqlcase::SqlCase::BuildInsertSqlListFromInput(int, std::vector<std::string, std::allocator<std::string> >*) const'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:411: undefined reference to `hybridse::sqlcase::SqlCase::ExtractInputTableDef(hybridse::sqlcase::SqlCase::TableInfo const&, hybridse::type::TableDef&) const'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:412: undefined reference to `hybridse::sqlcase::SqlCase::ExtractInputData(hybridse::sqlcase::SqlCase::TableInfo const&, std::vector<hybridse::codec::Row, std::allocator<hybridse::codec::Row> >&) const'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:451: undefined reference to `hybridse::sqlcase::SqlCase::ExtractOutputSchema(hybridse::type::TableDef&) const'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:456: undefined reference to `hybridse::sqlcase::SqlCase::ExtractOutputData(std::vector<hybridse::codec::Row, std::allocator<hybridse::codec::Row> >&) const'
libbase_test.a(sql_sdk_base_test.cc.o): In function `openmldb::sdk::SQLSDKQueryTest::BatchRequestExecuteSQLWithCommonColumnIndices(hybridse::sqlcase::SqlCase&, std::shared_ptr<openmldb::sdk::SQLRouter>, std::set<unsigned long, std::less<unsigned long>, std::allocator<unsigned long> > const&, bool, bool)':
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:559: undefined reference to `hybridse::sqlcase::SqlCase::ExtractTableDef(std::vector<std::string, std::allocator<std::string> > const&, std::vector<std::string, std::allocator<std::string> > const&, hybridse::type::TableDef&)'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:560: undefined reference to `hybridse::sqlcase::SqlCase::ExtractRows(google::protobuf::RepeatedPtrField<hybridse::type::ColumnDef> const&, std::vector<std::vector<std::string, std::allocator<std::string> >, std::allocator<std::vector<std::string, std::allocator<std::string> > > > const&, std::vector<hybridse::codec::Row, std::allocator<hybridse::codec::Row> >&)'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:563: undefined reference to `hybridse::sqlcase::SqlCase::ExtractInputTableDef(hybridse::sqlcase::SqlCase::TableInfo const&, hybridse::type::TableDef&) const'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:565: undefined reference to `hybridse::sqlcase::SqlCase::ExtractInputData(hybridse::sqlcase::SqlCase::TableInfo const&, std::vector<hybridse::codec::Row, std::allocator<hybridse::codec::Row> >&) const'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:611: undefined reference to `hybridse::sqlcase::SqlCase::ExtractOutputSchema(hybridse::type::TableDef&) const'
/root/OpenMLDB/src/sdk/sql_sdk_base_test.cc:615: undefined reference to `hybridse::sqlcase::SqlCase::ExtractOutputData(std::vector<hybridse::codec::Row, std::allocator<hybridse::codec::Row> >&) const'
libbase_test.a(base_test.cc.o): In function `openmldb::test::SQLCaseTest::InitCases(std::string const&)':
/root/OpenMLDB/src/test/base_test.cc:54: undefined reference to `hybridse::sqlcase::SqlCase::HYBRIDSE_LEVEL()'
libbase_test.a(base_test.cc.o): In function `openmldb::test::SQLCaseTest::InitCases(std::string const&, std::string const&, std::vector<hybridse::sqlcase::SqlCase, std::allocator<hybridse::sqlcase::SqlCase> >&)':
/root/OpenMLDB/src/test/base_test.cc:70: undefined reference to `hybridse::sqlcase::SqlCase::CreateSqlCasesFromYaml(std::string const&, std::string const&, std::vector<hybridse::sqlcase::SqlCase, std::allocator<hybridse::sqlcase::SqlCase> >&, std::vector<std::string, std::allocator<std::string> > const&)'
libbase_test.a(base_test.cc.o): In function `openmldb::test::SQLCaseTest::PrintSchema(google::protobuf::RepeatedPtrField<hybridse::type::ColumnDef> const&)':
/root/OpenMLDB/src/test/base_test.cc:90: undefined reference to `hybridse::sqlcase::SqlCase::TypeString(hybridse::type::Type)'
/root/OpenMLDB/src/test/base_test.cc:97: undefined reference to `hybridse::base::operator<<(std::ostream&, hybridse::base::TextTable const&)'
libbase_test.a(base_test.cc.o): In function `openmldb::test::SQLCaseTest::PrintRows(google::protobuf::RepeatedPtrField<hybridse::type::ColumnDef> const&, std::vector<hybridse::codec::Row, std::allocator<hybridse::codec::Row> > const&)':
/root/OpenMLDB/src/test/base_test.cc:214: undefined reference to `hybridse::base::operator<<(std::ostream&, hybridse::base::TextTable const&)'
libbase_test.a(base_test.cc.o): In function `openmldb::test::SQLCaseTest::PrintResultSet(std::shared_ptr<hybridse::sdk::ResultSet>)':
/root/OpenMLDB/src/test/base_test.cc:351: undefined reference to `hybridse::base::operator<<(std::ostream&, hybridse::base::TextTable const&)'
libbase_test.a(base_test.cc.o): In function `openmldb::test::SQLCaseTest::PrintResultSet(std::vector<std::shared_ptr<hybridse::sdk::ResultSet>, std::allocator<std::shared_ptr<hybridse::sdk::ResultSet> > >)':
/root/OpenMLDB/src/test/base_test.cc:394: undefined reference to `hybridse::base::operator<<(std::ostream&, hybridse::base::TextTable const&)'
collect2: error: ld returned 1 exit status
gmake[3]: *** [bin/sql_request_row_test] Error 1
gmake[3]: Leaving directory `/root/OpenMLDB/build'
gmake[2]: *** [src/sdk/CMakeFiles/sql_request_row_test.dir/all] Error 2
gmake[2]: Leaving directory `/root/OpenMLDB/build'
gmake[1]: *** [all] Error 2
gmake[1]: Leaving directory `/root/OpenMLDB/build'
make: *** [build] Error 2
```


**Expected Behavior**



**Steps to Reproduce**

1. 
2. 
3. 
4.

--- a/src/sdk/CMakeLists.txt
+++ b/src/sdk/CMakeLists.txt
@@ -17,36 +17,36 @@ if(TESTING_ENABLE)
     add_dependencies(mini_cluster_bm_common openmldb_proto hybridse_proto)
 
     add_library(base_test sql_sdk_base_test.cc ../test/base_test.cc)
-    add_dependencies(base_test openmldb_proto hybridse_proto)
+    add_dependencies(base_test hybridse_core openmldb_proto hybridse_proto)
 
-    set(HYBRIDSE_CASE_LIBS base_test yaml-cpp)
+    set(THIRD_LIBS benchmark_main benchmark ${GTEST_LIBRARIES} yaml-cpp)
 
     add_executable(db_sdk_test db_sdk_test.cc)
-    target_link_libraries(db_sdk_test ${BIN_LIBS} ${HYBRIDSE_CASE_LIBS} benchmark_main benchmark ${GTEST_LIBRARIES})
+    target_link_libraries(db_sdk_test base_test ${BIN_LIBS} ${THIRD_LIBS})
 
     add_executable(sql_router_test sql_router_test.cc)
-    target_link_libraries(sql_router_test ${BIN_LIBS} ${HYBRIDSE_CASE_LIBS} benchmark_main benchmark ${GTEST_LIBRARIES})
+    target_link_libraries(sql_router_test base_test ${BIN_LIBS} benchmark_main benchmark ${GTEST_LIBRARIES})
 
     add_executable(sql_standalone_sdk_test sql_standalone_sdk_test.cc)
-    target_link_libraries(sql_standalone_sdk_test ${GTEST_LIBRARIES} ${BIN_LIBS} ${HYBRIDSE_CASE_LIBS})
+    target_link_libraries(sql_standalone_sdk_test base_test ${BIN_LIBS} ${GTEST_LIBRARIES})
 
     add_executable(sql_sdk_test sql_sdk_test.cc)
-    target_link_libraries(sql_sdk_test ${GTEST_LIBRARIES} ${BIN_LIBS} ${HYBRIDSE_CASE_LIBS})
+    target_link_libraries(sql_sdk_test base_test ${BIN_LIBS} ${GTEST_LIBRARIES} )
 
     add_executable(sql_cluster_test sql_cluster_test.cc)
-    target_link_libraries(sql_cluster_test ${GTEST_LIBRARIES} ${BIN_LIBS} ${HYBRIDSE_CASE_LIBS})
+    target_link_libraries(sql_cluster_test base_test ${BIN_LIBS} ${GTEST_LIBRARIES})
 
     add_executable(sql_request_row_test sql_request_row_test.cc)
-    target_link_libraries(sql_request_row_test ${BIN_LIBS} ${HYBRIDSE_CASE_LIBS} ${ZETASQL_LIBS} benchmark_main benchmark ${GTEST_LIBRARIES})
+    target_link_libraries(sql_request_row_test base_test ${BIN_LIBS} ${ZETASQL_LIBS} ${THIRD_LIBS})
 
     add_executable(mini_cluster_batch_bm mini_cluster_batch_bm.cc)
-    target_link_libraries(mini_cluster_batch_bm mini_cluster_bm_common benchmark_main benchmark ${GTEST_LIBRARIES} ${BIN_LIBS} ${HYBRIDSE_CASE_LIBS})
+    target_link_libraries(mini_cluster_batch_bm mini_cluster_bm_common base_test ${BIN_LIBS} ${THIRD_LIBS})
 
     add_executable(mini_cluster_request_batch_bm mini_cluster_request_batch_bm.cc)
-    target_link_libraries(mini_cluster_request_batch_bm mini_cluster_bm_common benchmark_main benchmark ${GTEST_LIBRARIES} ${BIN_LIBS} ${HYBRIDSE_CASE_LIBS})
+    target_link_libraries(mini_cluster_request_batch_bm mini_cluster_bm_common base_test ${BIN_LIBS} ${THIRD_LIBS})
 
     add_executable(mini_cluster_request_bm mini_cluster_request_bm.cc)
-    target_link_libraries(mini_cluster_request_bm mini_cluster_bm_common benchmark_main benchmark ${GTEST_LIBRARIES} ${BIN_LIBS} ${HYBRIDSE_CASE_LIBS})
+    target_link_libraries(mini_cluster_request_bm mini_cluster_bm_common base_test ${BIN_LIBS} ${THIRD_LIBS})
 endif()
 
 set(SDK_LIBS openmldb_sdk openmldb_catalog client zk_client schema openmldb_flags openmldb_codec openmldb_proto base hybridse_sdk zookeeper_mt)

